testutil/daemon/daemon.go

```diff
@@ -253,16 +256,22 @@ func (d *Daemon) StartWithLogFile(out *os.File, providedArgs ...string) error {
 	}
 
 	d.args = append(d.GlobalFlags,
-		"--containerd", containerdSocket,
 		"--data-root", d.Root,
 		"--exec-root", d.execRoot,
 		"--pidfile", d.pidFile,
 		fmt.Sprintf("--userland-proxy=%t", d.userlandProxy),
 		"--containerd-namespace", d.id,
 		"--containerd-plugins-namespace", d.id+"p",
 	)
+	if d.containerdSocket != "" {
+		d.args = append(d.args, "--containerd", d.containerdSocket)
+	}
+	if d.noIptables {
+		d.args = append(d.args, "--iptables=false")
```

Comment:
@SamWhited I don't believe you need this anymore (this was needed when run inside buildkit's alpine container but that's no longer the design we're taking). Also if it'd be needed it could be passed to `providedArgs`. The reason we needed an option for containerd was because a default was hardcoded and there was no way to change it.

Reply:
Good pint; I'll test without it.